finagle-core/src/main/scala/com/twitter/finagle/loadbalancer/flags.scala

```diff
@@ -41,6 +41,14 @@ object perHostStats
         "\tor the NullStatsReceiver if none given."
     )
 
+object useCanonicalHostname
```

Comment:
questions: any reason we want to use a flag instead of stack param. perhaps an `AddressFormatter` stack param that can be invoked on an address?